resources/assets/js/components/PageList.vue

```diff
@@ -1,45 +1,135 @@
 <template>
 <div>
 	<back-bar :title="title" />
-	<ul v-if="pages" v-for="page in pages" class="page-list">
-		<PageListItem class="item" :site="site" :page="page" :editing="edit"></PageListItem>
-	</ul>
+
+	<div v-if="pages" v-for="page in pages" class="page-list">
+		<page-list-item
+			:key="page.id"
+			:page="page"
+			:flatten="true"
+			:open-modal="openModal"
+			path="0"
+			:depth="0"
+		/>
+	</div>
+
+	<!-- TODO: move out into a container fo all modal windows -->
+	<div id="create-form">
+		<el-dialog title="Create Page" v-if="createFormVisible" :visible.sync="createFormVisible" :modal-append-to-body=false >
+			<el-form :model="createForm">
+				<el-form-item label="Page title">
+					<el-input name="title" v-model="createForm.title" auto-complete="off"></el-input>
+				</el-form-item>
+				<el-select name="layout_name" v-model="createForm.layout_name" placeholder="Select">
+					<el-option
+							v-for="item in layouts"
+							:key="item"
+							:label="item"
+							:value="item">
+					</el-option>
+				</el-select>
+				<el-form-item label="Layout Version">
+					<el-input name="layout_version" v-model="createForm.layout_version" auto-complete="off"></el-input>
+				</el-form-item>
+				<el-form-item label="slug">
+					<el-input name="slug" v-model="createForm.route.slug" auto-complete="off"></el-input>
+				</el-form-item>
+			</el-form>
+			<span slot="footer" class="dialog-footer">
+			<el-button @click="createFormVisible = false">Cancel</el-button>
+			<el-button type="primary" @click="addChild">Confirm</el-button>
+		</span>
+		</el-dialog>
+	</div>
+
+	<div class="b-bottom-bar">
+		<el-button class="u-mla" @click="() => { this.openModal(this.$children[2]) }">+ Add Page</el-button>
```

Comment:
Directly referring to the children like this feels yucky but is good enough for now.